INSTALL.md

```diff
@@ -0,0 +1,171 @@
+
+Starting a new stack project is as simple as running `stack init`.
+Stack should now inform you that it will use the relevant `stack` and
+`cabal` files from `xmonad` and `xmonad-contrib` to generate its
+`stack.yaml` file.  At the time of writing, this looks a little bit like
+this:
+
+```
+  $ stack init
+  Looking for .cabal or package.yaml files to use to init the project.
+  Using cabal packages:
+  - xmonad-contrib/
+  - xmonad/
+
+  Selecting the best among 19 snapshots...
+
+  * Matches https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/17/9.yaml
+
+  Selected resolver: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/17/9.yaml
+  Initialising configuration using resolver: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/17/9.yaml
+  Total number of user packages considered: 2
+  Writing configuration to file: stack.yaml
+  All done.
+```
+
+If you look into your current directory now, you should see a freshly
+generated `stack.yaml` file:
+
+```
+  $ ls
+  xmonad  xmonad-contrib  stack.yaml  xmonad.hs
+```
+
+The meat of that file (comments start with `#`, we've omitted them here)
+will look a little bit like
+
+``` yaml
+  resolver:
+    url: https://raw.githubusercontent.com/commercialhaskell/stackage-snapshots/master/lts/17/9.yaml
+
+  packages:
+  - xmonad
+  - xmonad-contrib
+```
+
+### Install Everything
+
+Installing things is as easy as typing `stack install`.  This will
+install the correct version of GHC, as well as build all of the required
+packages (`stack build`) and then copy the relevant executables
+(`xmonad`, in our case) to `~/.local/bin`.  Make sure to add that
+directory to your `$PATH`!
+
+### Tell XMonad How to Recompile Itself
+
+In order to tell xmonad to invoke `stack build` when we issue `xmonad
+--recompile` (bound to `M-q` by default), we need to create a so-called
+`build` file.  This is quite literally just a shell script called
+`build` in your xmonad directory (which is `~/.config/xmonad` for us)
+that tells xmonad how it should build its executable.
+
+A good starting point (this is essentially [what xmonad would do]
+without a build file, with the exception that we are invoking `stack
+ghc` instead of plain `ghc`) would be
+
+``` shell
+  #!/bin/sh
+
+  exec stack ghc --  \
+    --make xmonad.hs \
+    -i               \
+    -ilib            \
+    -fforce-recomp   \
+    -main-is main    \
+    -v0              \
+    -o "$1"
+```
+
+Don't forget to mark the file as `+x`: `chmod +x build`!
+
+And that's it!  Recompilation should work normally now, though you will
+potentially need to restart your computer, or at least the running X
+session, first.
+
+### Don't Recompile on Every Startup
+
+By default, xmonad always recompiles itself when a build script is used
+(because the build script could contain arbitrary code, so a simple
+check whether the `xmonad.hs` file changed is not enough).  If you find
```

Comment:
I am a bit confused by that. I think usually users will run `xmonad --recompile`, and then xmonad will recognize the build script and recompile. But this sentence may sound like every time `xmonad` starts, it will recompile. Maybe a bit clarification.

Reply:
It sounds like I'm saying that because I'm saying exactly that! :)  When you use a build script we have no idea what it does and so we recompile _every time_ xmonad starts